packages/cozy-harvest-lib/src/models/ConnectionFlow.js

```diff
@@ -617,20 +655,45 @@ export class ConnectionFlow {
    * If the trigger we display is already running, subscribe to the associated job
    */
   watchCurrentJobIfTriggerIsAlreadyRunning() {
-    if (get(this, 'trigger.current_state.status') === 'running') {
+    if (this.trigger?.current_state?.status === 'running') {
       this.job = {
-        _id: get(this, 'trigger.current_state.last_executed_job_id')
+        _id: this.trigger?.current_state?.last_executed_job_id
       }
       this.watchJob()
     }
   }
 
+  /**
+   * Watch all jobs related to the current trigger
+   */
+  watchTriggerJobs() {
+    if (this.trigger) {
+      // When the trigger comes from realtime, cozy-stack does not add the current_state to the object. So we need to request the stack to get it
+      if (!this.trigger?.current_state) {
+        this.refetchTrigger()
+      }
+      this.realtime.subscribe(
+        'updated',
+        JOBS_DOCTYPE,
+        this.handleTriggerJobUpdated
+      )
+    }
+  }
+
   watchJob(options = { autoSuccessTimer: false }) {
+    if (this.job?._id === this.jobWatcher?.job?._id) {
+      // no need to rewatch a job we are already watching
+      return
+    }
```

Comment:
We should not call `watchJob` if the job is already watched. This should be the real fix.

Reply:
`Watchjob` is called from `watchCurrentJobIfTriggerIsAlreadyRunning` and from `launch`. In the case of launch, we call `watchjob` directly with the job we received directly from the trigger launch which in this case is in double with `watchCurrentjobIfTriggerIsAlreadyRunning`.

I could remove the call to `watchjob` in `launch` but that would make `watchjob` totally dependent of realtime which I think is not a good solution.